odoo/addons/base/models/ir_qweb_fields.py

```diff
@@ -609,17 +609,25 @@ def get_available_options(self):
             width=dict(type='integer', string=_('Width'), default_value=600),
             height=dict(type='integer', string=_('Height'), default_value=100),
             humanreadable=dict(type='integer', string=_('Human Readable'), default_value=0),
+            quiet=dict(type='integer', string='Quiet', default_value=1),
+            mask=dict(type='string', string='Mask', default_value='')
         )
         return options
 
     @api.model
     def value_to_html(self, value, options=None):
+        if not value:
+            return ''
         barcode_symbology = options.get('symbology', 'Code128')
         barcode = self.env['ir.actions.report'].barcode(
             barcode_symbology,
             value,
-            **{key: value for key, value in options.items() if key in ['width', 'height', 'humanreadable']})
-        return u'<img src="data:png;base64,%s">' % base64.b64encode(barcode).decode('ascii')
+            **{key: value for key, value in options.items() if key in ['width', 'height', 'humanreadable', 'quiet', 'mask']})
```

Comment:
Neither `quiet` nor `mask` seem to be valid parameters for `ir.actions.report.barcode()` in 12.0.
Considering they were introduced by various intermediary commits, do we really need to backport these as well in 12.0 where they were not supported by the `/report/barcode` either in 12.0?